core/src/test/java/org/elasticsearch/search/aggregations/InternalMultiBucketAggregationTestCase.java

```diff
@@ -57,7 +72,10 @@ public void setUp() throws Exception {
 
     @Override
     protected final T createTestInstance(String name, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) {
-        return createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+        T instance = createTestInstance(name, pipelineAggregators, metaData, subAggregationsSupplier.get());
+        assert instance.getBuckets().size() <= maxNumberOfBuckets() :
```

Comment:
small suggestion: instead of having to verify this in the base class, how about having an abstract createBucket method that gets called as many times as needed from the base class? Then we could even think of dropping the minNumberOfBuckets and minNumberOfBuckets methods for simplicity and have subclasses override randomNumberOfBuckets only?
Leave this out if it makes things too complicated rather than simplifying. I think it would be cleaner but I am not sure how easy of a change it would be to make.